osvimdriver/openstack/heat/driver.py

```diff
@@ -29,29 +34,71 @@ def create_stack(self, stack_name, heat_template, input_properties=None, files=N
             raise ValueError('heat_template must be provided')
         heat_client = self.__get_heat_client()
         logger.debug('Creating stack with name %s', stack_name)
-        create_result = heat_client.stacks.create(stack_name=stack_name, template=heat_template, parameters=input_properties, files=files)
-        stack_id = create_result['stack']['id']
-        logger.debug('Stack with name %s created and assigned id %s', stack_name, stack_id)
-        return stack_id
 
-    def delete_stack(self, stack_id):
+        external_request_id = str(uuid.uuid4())
+
+        reqbody_dict = {"stack_name" : stack_name, "template" : heat_template, "parameters" : input_properties, "files" : files}
+        self._generate_additional_logs(reqbody_dict, 'sent', external_request_id, 'application/json',
+                                       'request', 'http', {'method' : 'post', 'uri' : LOG_URI_PREFIX +'/stacks'}, driver_request_id)
+        try:
+            create_result = heat_client.stacks.create(stack_name=stack_name, template=heat_template, parameters=input_properties, files=files)
+            stack_id = create_result['stack']['id']
+            driver_request_id = rd.build_request_id(rd.CREATE_REQUEST_PREFIX, str(stack_id))
```

Comment:
This driver_request_id must be returned to the caller in @ https://github.com/IBM/openstack-vim-driver/blob/develop/osvimdriver/service/resourcedriver.py. at line number 166, remove creating the driver_request_id again in line number 167, the retuned value should be passed in line number 168.

Reply:
I have done changes.